src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  useContext,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import classNames from 'classnames';
+
+import { Context } from '../../Context';
+
+import { Todo } from '../../types/Todo';
+import { deleteTodo, updateTodo } from '../../api/todos';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+  const {
+    handleErrorChange,
+    setTodos,
+  } = useContext(Context);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [editing, setEditing] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState(title);
+
+  inputRef.current?.focus();
+
+  const handleEdit = () => {
+    setEditing(true);
+
+    if (inputRef.current) {
+      inputRef.current.value = title;
+    }
+  };
+
+  const handleUpdateTodo = (updatedTodo: Todo) => {
+    setIsLoading(true);
```

Comment:
Would be nice to show the loader when you delete/add/delete all completed/toggle all todos, as in the working example.
It's not required in the task (except adding) so optionally - just if you are able to do it.

About adding: in the previous task description was the instruction with `tempTodo` about that:
https://github.com/mate-academy/react_todo-app-add-and-delete#adding-a-todo